#166265828 Initialize express server

chuxmykel opened the pull request:
#### What does this PR do?
This PR initializes the express server and updates the package.json with relevant scripts

#### Description of Task to be completed?
- install dotenv and update server to run on port gotten from env variables
- add a .env.sample file to the repo
- add dev and start scripts for development and production environments
- create server folder and move all server related filed to the folder
- update README.md file with instructions on how to run the project

#### How should this be manually tested?
- clone the repo and switch to the repo directory
- run `npm run build` and then `npm run start` to start the project in production mode or run `npm run dev` to run it in development mode
#### Any background context you want to provide?
- None
#### What are the relevant pivotal tracker stories?
- [#166265828](https://www.pivotaltracker.com/story/show/166265828)

#### Screenshots (if appropriate)
![Screenshot from 2019-05-27 18-24-41](https://user-images.githubusercontent.com/44305641/58432970-e2d74600-80ac-11e9-95aa-63f44ce7c6b5.png)
![Screenshot from 2019-05-27 18-22-03](https://user-images.githubusercontent.com/44305641/58432971-e36fdc80-80ac-11e9-94a7-28ee8187fe0a.png)

chuxmykel committed:
chore(server setup): initialize express server with environment variables

- install dotenv and update server to run on port gotten from env variables
- add dev and start scripts for development and production environments
- create server folder and move all server related filed to the folder
- update README.md file with instructions on how to run the project

[Finishes #166265828]

emp-daisy commented:
Ensure you invite your teammates to review first

chuxmykel commented:
All right. Teammates review requested. Thanks

OvieMudi commented on package.json:
```diff
@@ -4,8 +4,12 @@
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "start": "node ./build/index.js",
+    "dev": "nodemon --exec babel-node server/index.js",
```
Great job. I only have an issue with the npm script.
Would have preferred the conventional `startDev`  or  `devStart`naming, instead of just "dev".

encodedBicoding replied:
Do we go with the `npm run dev` or `npm run startDev` ?

PelumiAlesh commented on server/index.js:
```diff
@@ -1,8 +1,11 @@
-const port = 3000;
+const port = process.env.PORT || 3000;
 
 app.get('/', (req, res) => res.send('This is Hello Books'));
```
The acceptance criteria stated that a wild card route to handle **any endpoint**.
_A wildcard route to handle any endpoints, to return a JSON message { "message": "Hello Books Deferral" }_
```suggestion
app.get('/*', (req, res) => res.send('Hello Books Deferral'));
```

ocranbillions commented on README.md:
````diff
@@ -1,4 +1,44 @@
+
+```shell
+git clone https://github.com/emp-daisy/HelloBooksLib.git
+cd banka/
````
@chuxmykel this obviously isn't meant to be here right?

chuxmykel replied:
@ocranbillions what exactly?

chuxmykel replied:
Oh. Now I see it. I'll fix it. Thanks.

chuxmykel committed:
chore(server setup): initialize express server with environment variables

-install dotenv and update server to run on port gotten from env variables
-add dev and start scripts for development and production environments
-create server folder and move all server related filed to the folder
-update README.md file with instructions on how to run the project

[Finishes #166265828]

Hadeneekeh reviewed (dismissed):
Well done @chuxmykel

emp-daisy commented:
Please resolve conflicts

chuxmykel committed:
Merge branch 'develop' into ch/166265828/initialize-env-variables

emp-daisy commented on .env.sample:
```diff
@@ -0,0 +1 @@
+PORT = integer
```
Make this more descriptive

chuxmykel replied:
All right. Done

chuxmykel committed:
chore(server setup): initialize express server with environment variables
-install dotenv and update server to run on port gotten from env variables
-add dev and start scripts for development and production environments
-create server folder and move all server related filed to the folder
-update README.md file with instructions on how to run the project
[Finishes #166265828]

OvieMudi reviewed (approved):
All seems fine to me.

ocranbillions reviewed (approved):
Looks ok to me as well.
